### DISCUSSION TITLE
rosbag support

### DISCUSSION THREAD

**Discussion by ahans** [mvukov/rules_ros2#26] [General]
Hey, how am I supposed to record rosbags with custom types? With a "standard" ROS installed alongside it doesn't work, since ROSBAG2_TRANSPORT complains about unknown types. Briefly checking the code it looks like it wants something like `libmytypes_msgs__rosidl_typesupport_cpp.so`. Are there plans on how to support something like that?

**Comment by mvukov** [comment]
Hello, I am working on a rosbag support. To answer your first question: I need to figure this out yet :)

**Reply by ahans** [reply]
Do you have your current state on a branch somewhere? Maybe I could help. I think a good approach would be one that is similar to what you did for `ros2 topic` support. Then we could have a target that accepts a list of types and runs `ros2 bag` in a way that they are available.

**Reply by mvukov** [reply]
Looks like rosbag stuff heavily depends on pluginlib (and therefore on ament_index_cpp). I plan to work out some Bazel functionality to create a file structure such that ament_index_cpp can find what pluginlib asks for. We'll probably also need shared libs, like you mentioned `libmytypes_msgs__rosidl_typesupport_cpp.so`:  we already have C++ IDL libraries, the job would be to extract the shared library. Please share your ideas. :)

Once that works, pluginlib and rosbag should work :crossed_fingers:. The current work in in [feature/pluginlib](https://github.com/mvukov/rules_ros2/tree/feature/pluginlib) branch. Eventually, it would be nice to have a high level target/macro `ros2_bag`.

**Reply by mvukov** [reply]
rosbag stuff is getting into shape. It requires a bit more work to become functional.

**Reply by mvukov** [reply]
In #47 we added rosbag2 support.

**Comment by thomasegriffith** [comment]
For vis, see #50 for another approach
